src/main/ruby/truffleruby/core/match_data.rb

```diff
@@ -61,6 +66,25 @@ def string
   def captures
     to_a[1..-1]
   end
+  alias_method :deconstruct, :captures
+
+  def deconstruct_keys(array_of_names)
+    Truffle::Type.rb_check_type(array_of_names, Array) unless Primitive.nil?(array_of_names)
+
+    hash = named_captures.transform_keys(&:to_sym)
+    return hash if Primitive.nil?(array_of_names)
+
+    ret = {}
+    return ret if array_of_names.size > hash.size
+
+    array_of_names.each do |key|
+      Truffle::Type.rb_check_type(key, Symbol)
+      break unless hash.key?(key)
+      ret[key] = hash[key]
```

Comment:
I considered
```suggestion
      ret[key] = hash.fetch(key) { return ret }
```
I dislike doing the key lookup twice, but I'm not sure if that is better than introducing a block.

Reply:
There is `Primitive.hash_get_or_undefined` which could be used for that, or `hash.fetch(key, undefined)`.

Reply:
It makes sense to use `Primitive.hash_get_or_undefined` here, I or Andrii will do it to avoid an extra round-trip.